pkg/util/helper/cache_test.go

```diff
@@ -0,0 +1,304 @@
+package helper
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/keys"
+)
+
+func TestGetObjectFromCache(t *testing.T) {
+	type args struct {
+		restMapper meta.RESTMapper
+		manager    func(<-chan struct{}) genericmanager.MultiClusterInformerManager
+		fedKey     keys.FederatedKey
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *unstructured.Unstructured
+		wantErr bool
+	}{
+		{
+			name: "kind not registered",
+			args: args{
+				restMapper: meta.NewDefaultRESTMapper(nil),
+				manager: func(stopCh <-chan struct{}) genericmanager.MultiClusterInformerManager {
+					return genericmanager.NewMultiClusterInformerManager(stopCh)
+				},
+				fedKey: keys.FederatedKey{Cluster: "cluster", ClusterWideKey: keys.ClusterWideKey{
+					Version: "v1", Kind: "Pod", Namespace: "default", Name: "pod",
+				}},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "cluster manager not found",
+			args: args{
+				restMapper: func() meta.RESTMapper {
+					m := meta.NewDefaultRESTMapper([]schema.GroupVersion{corev1.SchemeGroupVersion})
+					m.Add(schema.GroupVersionKind{Version: "v1", Kind: "Pod"}, meta.RESTScopeNamespace)
+					return m
+				}(),
+				manager: func(stopCh <-chan struct{}) genericmanager.MultiClusterInformerManager {
```

Comment:
the way to create `restMapper`  `manager` is quite similar, can we extract it as a common function?